### PR TITLE
Install helper to setup CoCo on baremetal worker nodes

### PR DESCRIPTION
**Pull request by bpradipt**
Setup CoCo on baremetal OpenShift worker nodes using the layered image deployment feature of OSC operator.

Added few additional commits to move the scripts from hack to scripts folder

### CODE REVIEW

**Comment by bpradipt** [comment]
@apinnick I have moved the scripts from hack folder to scripts folder in this PR. PTAL

**Review by apinnick — APPROVED** [review]
LGTM

File: scripts/install-helpers/baremetal-coco/.gitignore [hunk 1]
```diff
@@ -0,0 +1,4 @@
+id_rsa*
+*kubeconfig*
```
Comment (gkurz) [inline scripts/install-helpers/baremetal-coco/.gitignore]: The `install.sh` script doesn't deal with kubeconfig or id_rsa files in any way AFAICT. Why is this needed ?

File: scripts/install-helpers/baremetal-coco/README.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+NodeFeatureDiscovery (NFD) operator is used to label the TDX and SNP nodes.
+`intel.feature.node.kubernetes.io/tdx: "true"` is used for TDX nodes and
+`amd.feature.node.kubernetes.io/snp: "true"` is used for SNP nodes.
+
+Kata runtime is configured on the nodes with the above labels.
+Note that currently the script only supports installing a single TEE environment.
+
+## Prerequisites
+
```
Comment (gkurz) [inline scripts/install-helpers/baremetal-coco/README.md:14]: You might want to list the `oc` and `jq` commands there since they are required by `install.sh`.

File: scripts/install-helpers/baremetal-coco/install.sh [hunk 3]
```diff
@@ -0,0 +1,733 @@
...
+            echo "MCP $mcp is ready"
+            return 0
+        fi
+        sleep $interval
+        elapsed=$((elapsed + interval))
+        statusUpdated=$(oc get mcp "$mcp" -o=jsonpath='{.status.conditions[?(@.type=="Updated")].status}')
+        statusUpdating=$(oc get mcp "$mcp" -o=jsonpath='{.status.conditions[?(@.type=="Updating")].status}')
+        statusDegraded=$(oc get mcp "$mcp" -o=jsonpath='{.status.conditions[?(@.type=="Degraded")].status}')
+    done
+
```
Comment (gkurz) [inline scripts/install-helpers/baremetal-coco/install.sh]:
```suggestion
    echo "MCP $mcp is not ready after $timeout seconds"
    return 1
```

File: scripts/install-helpers/baremetal-coco/install.sh [hunk 4]
```diff
@@ -0,0 +1,733 @@
...
+    local timeout=$CMD_TIMEOUT
+    local interval=5
+    local elapsed=0
+    local ready=0
+
+    # oc get runtimeclass "$runtimeclass" -o jsonpath={.metadata.name} should return the runtimeclass
+    while [ $elapsed -lt "$timeout" ]; do
+        ready=$(oc get runtimeclass "$runtimeclass" -o jsonpath='{.metadata.name}')
+        if [ "$ready" == "$runtimeclass" ]; then
+            echo "Runtimeclass $runtimeclass is ready"
```
Comment (gkurz) [inline scripts/install-helpers/baremetal-coco/install.sh]:
```suggestion
            echo "RuntimeClass $runtimeclass is ready"
```

File: scripts/install-helpers/baremetal-coco/install.sh [hunk 5]
```diff
@@ -0,0 +1,733 @@
...
+    local namespace=$2
+    local timeout=$CMD_TIMEOUT
+    local interval=5
+    local elapsed=0
+    local ip=0
+
+    while [ $elapsed -lt "$timeout" ]; do
+        ip=$(oc get endpoints -n "$namespace" "$service" -o jsonpath='{.subsets[0].addresses[0].ip}')
+        if [ -n "$ip" ]; then
+            echo "Service $service IP is available"
```
Comment (gkurz) [inline scripts/install-helpers/baremetal-coco/install.sh]: Maybe include `"$ip"` in the output ?

File: scripts/install-helpers/baremetal-coco/install.sh [hunk 6]
```diff
@@ -0,0 +1,733 @@
...
+        ready=$(oc get runtimeclass "$runtimeclass" -o jsonpath='{.metadata.name}')
+        if [ "$ready" == "$runtimeclass" ]; then
+            echo "Runtimeclass $runtimeclass is ready"
+            return 0
+        fi
+        sleep $interval
+        elapsed=$((elapsed + interval))
+    done
+
+    echo "Runtimeclass $runtimeclass is not ready after $timeout seconds"
```
Comment (gkurz) [inline scripts/install-helpers/baremetal-coco/install.sh]:
```suggestion
    echo "RuntimeClass $runtimeclass is not ready after $timeout seconds"
```

File: scripts/install-helpers/baremetal-coco/install.sh [hunk 7]
```diff
@@ -0,0 +1,733 @@
...
+    return 1
+}
+
+# Function to apply the operator manifests
+function apply_operator_manifests() {
+    # Apply the manifests, error exit if any of them fail
```
Comment (gkurz) [inline scripts/install-helpers/baremetal-coco/install.sh]: What is the motivation to `exit 1` instead of `return 1` ?

File: scripts/install-helpers/baremetal-coco/install.sh [hunk 8]
```diff
@@ -0,0 +1,733 @@
...
+
+# Function to set additional cluster-wide image pull secret
+# Requires PULL_SECRET_JSON environment variable to be set
+# eg. PULL_SECRET_JSON='{"my.registry.io": {"auth": "ABC"}}'
+function add_image_pull_secret() {
+    # Check if SECRET_JSON is set
+    if [ -z "$PULL_SECRET_JSON" ]; then
+        echo "PULL_SECRET_JSON environment variable is not set"
+        echo "example PULL_SECRET_JSON='{\"my.registry.io\": {\"auth\": \"ABC\"}}'"
+        exit 1
```
Comment (gkurz) [inline scripts/install-helpers/baremetal-coco/install.sh]: Same question as with `apply_operator_manifests()`.

File: scripts/install-helpers/baremetal-coco/install.sh [hunk 9]
```diff
@@ -0,0 +1,733 @@
...
+    wait_for_service_ep_ip webhook-service openshift-sandboxed-containers-operator || return 1
+
+    echo "OpenShift sandboxed containers operator | deployment finished successfully"
+}
+
+# Function to deploy NodeFeatureDiscovery (NFD) operator
+function deploy_nfd_operator() {
+    echo "Node Feature Discovery operator | starting the deployment"
+
+    pushd nfd || return 1
```
Comment (gkurz) [inline scripts/install-helpers/baremetal-coco/install.sh:218]: Since the `nfd` directory is a strong requirement here, I'd rather check its existence upfront, before calling this function and explicitly report when it is missing.

File: scripts/install-helpers/baremetal-coco/install.sh [hunk 10]
```diff
@@ -0,0 +1,733 @@
...
+    oc apply -f ns.yaml || return 1
+    oc apply -f og.yaml || return 1
+    oc apply -f subs.yaml || return 1
+    popd || return 1
```
Comment (gkurz) [inline scripts/install-helpers/baremetal-coco/install.sh:222]: The only reason for `popd` to fail would be the removal of the current working directory at the time `pushd` was called... can this even happen ?

File: scripts/install-helpers/baremetal-coco/install.sh [hunk 11]
```diff
@@ -0,0 +1,733 @@
...
+    wait_for_deployment nfd-controller-manager openshift-nfd || return 1
+    echo "Node Feature Discovery operator | deployment finished successfully"
+}
+
+function create_intel_node_feature_rules() {
+    echo "Node Feature Discovery operator | creating intel node feature rules"
+
+    pushd nfd || return 1
+    oc apply -f intel-rules.yaml || return 1
+    popd || return 1
```
Comment (gkurz) [inline scripts/install-helpers/baremetal-coco/install.sh:233]: Same comments as with `deploy_nfd_operator()`.

File: scripts/install-helpers/baremetal-coco/install.sh [hunk 12]
```diff
@@ -0,0 +1,733 @@
...
+
+function deploy_intel_device_plugins() {
+    echo "Intel Device Plugins operator | starting the deployment"
+
+    pushd intel-dpo || return 1
+    oc apply -f install_operator.yaml || return 1
+    wait_for_deployment inteldeviceplugins-controller-manager openshift-operators || return 1
+    oc apply -f sgx_device_plugin.yaml || return 1
+    popd || return 1
+
```
Comment (gkurz) [inline scripts/install-helpers/baremetal-coco/install.sh:246]: Same comments as with deploy_nfd_operator().

File: scripts/install-helpers/baremetal-coco/install.sh [hunk 13]
```diff
@@ -0,0 +1,733 @@
...
+    echo "Intel Device Plugins operator | deployment finished successfully"
+}
+
+function create_amd_node_feature_rules() {
+    echo "Node Feature Discovery operator | creating amd node feature rules"
+
+    pushd nfd || return 1
+    oc apply -f amd-rules.yaml || return 1
+    popd || return 1
+
```
Comment (gkurz) [inline scripts/install-helpers/baremetal-coco/install.sh:256]: Same comments as with deploy_nfd_operator().

File: scripts/install-helpers/baremetal-coco/install.sh [hunk 14]
```diff
@@ -0,0 +1,733 @@
...
+    case $tee_type in
+    tdx)
+        filepath=/etc/kata-containers/tdx/config.d/96-kata-kernel-config
+        ;;
+    snp)
+        filepath=/etc/kata-containers/snp/config.d/96-kata-kernel-config
+        ;;
+    esac
+
+    cat <<EOF >./96-kata-kernel-config-mc.yaml
```
Comment (gkurz) [inline scripts/install-helpers/baremetal-coco/install.sh]: This literal string as 3 other users. Maybe put it in a global ?

Also you might want to add this file to `.gitignore`.

File: scripts/install-helpers/baremetal-coco/install.sh [hunk 15]
```diff
@@ -0,0 +1,733 @@
...
+  kataConfigPoolSelector:
+    matchLabels:
+      $label
+EOF
+
+    result=$?
+    [ $result -eq 0 ] || return 1
+
+    echo "KataConfig object successfully created"
+    return 0
```
Comment (gkurz) [inline scripts/install-helpers/baremetal-coco/install.sh]: Not needed.
```suggestion
```

File: scripts/install-helpers/baremetal-coco/install.sh [hunk 16]
```diff
@@ -0,0 +1,733 @@
...
+    tdx)
+        label='intel.feature.node.kubernetes.io/tdx: "true"'
+        ;;
+    snp)
+        label='amd.feature.node.kubernetes.io/snp: "true"'
+        ;;
+    esac
+
+    # Create KataConfig object
+    oc apply -f - <<EOF
```
Comment (gkurz) [inline scripts/install-helpers/baremetal-coco/install.sh]: If you do :
```suggestion
    oc apply -f - || return 1 <<EOF
```
you can...

File: scripts/install-helpers/baremetal-coco/install.sh [hunk 17]
```diff
@@ -0,0 +1,733 @@
...
+spec:
+  enablePeerPods: false
+  logLevel: info
+  kataConfigPoolSelector:
+    matchLabels:
+      $label
+EOF
+
+    result=$?
+    [ $result -eq 0 ] || return 1
```
Comment (gkurz) [inline scripts/install-helpers/baremetal-coco/install.sh]: ... drop these lines
```suggestion
```

File: scripts/install-helpers/baremetal-coco/install.sh [hunk 18]
```diff
@@ -0,0 +1,733 @@
...
+      - contents:
+          source: data:text/plain;charset=utf-8;base64,$source
+        mode: 420
+        overwrite: true
+        path: $filepath
+EOF
+
+    oc apply -f 96-kata-kernel-config-mc.yaml || return 1
+
+}
```
Comment (gkurz) [inline scripts/install-helpers/baremetal-coco/install.sh]: The `|| return 1` isn't needed.

```suggestion
    oc apply -f 96-kata-kernel-config-mc.yaml
}
```

File: scripts/install-helpers/baremetal-coco/install.sh [hunk 19]
```diff
@@ -0,0 +1,733 @@
...
+        ;;
+    snp)
+        ext_resources='sev-snp.amd.com/esids: 1'
+        ;;
+    esac
+
+    echo "Creating kata-$tee_type RuntimeClass object"
+
+    #Create runtimeClass object
+    oc apply -f - <<EOF
```
Comment (gkurz) [inline scripts/install-helpers/baremetal-coco/install.sh]: If you do :
```suggestion
    oc apply -f - || return 1 <<EOF
```
you can...

File: scripts/install-helpers/baremetal-coco/install.sh [hunk 20]
```diff
@@ -0,0 +1,733 @@
...
+  podFixed:
+    memory: "350Mi"
+    cpu: "250m"
+    $ext_resources
+scheduling:
+  nodeSelector:
+     $label
+EOF
+    result=$?
+    [ $result -eq 0 ] || return 1
```
Comment (gkurz) [inline scripts/install-helpers/baremetal-coco/install.sh]: ... drop these lines
```suggestion
```

File: scripts/install-helpers/baremetal-coco/install.sh [hunk 21]
```diff
@@ -0,0 +1,733 @@
...
+
+    echo "kata-$tee_type RuntimeClass object successfully created"
+    return 0
+
+}
```
Comment (gkurz) [inline scripts/install-helpers/baremetal-coco/install.sh]: Not needed.
```suggestion
}
```

File: scripts/install-helpers/baremetal-coco/install.sh [hunk 22]
```diff
@@ -0,0 +1,733 @@
...
+# Function to verify all required variables are set and
+# required files exist
+
+function verify_params() {
+
+    # Check if TEE_TYPE is provided
+    if [ -z "$TEE_TYPE" ]; then
+        echo "Error: TEE type (-t) is mandatory"
+        display_help
+        exit 1
```
Comment (gkurz) [inline scripts/install-helpers/baremetal-coco/install.sh]: What about `return 1` and let the caller decide ?

File: scripts/install-helpers/baremetal-coco/install.sh [hunk 23]
```diff
@@ -0,0 +1,733 @@
...
+
+}
+
+function uninstall_node_feature_discovery() {
+    tee_type="${1:-}"
+
+    oc get deployment nfd-controller-manager -n openshift-nfd &>/dev/null
+    return_code=$?
+    if [ $return_code -eq 0 ]; then
+        pushd nfd || return 1
```
Comment (gkurz) [inline scripts/install-helpers/baremetal-coco/install.sh:482]: Same remark as with other invocations `pushd` and `popd`.

File: scripts/install-helpers/baremetal-coco/install.sh [hunk 24]
```diff
@@ -0,0 +1,733 @@
...
+    fi
+}
+
+# Function to uninstall the installed artifacts
+# It won't delete the cluster
+function uninstall() {
+    echo "Uninstalling all the artifacts"
+
+    # Uninstall NFD
+    uninstall_node_feature_discovery "$TEE_TYPE" || exit 1
```
Comment (gkurz) [inline scripts/install-helpers/baremetal-coco/install.sh]: What about `return 1` and let the caller decide ?

File: scripts/install-helpers/baremetal-coco/install.sh [hunk 25]
```diff
@@ -0,0 +1,733 @@
...
+oc apply -f osc-fg-cm.yaml || exit 1
+
+# Create Layered Image FG ConfigMap
+if [ "$TEE_TYPE" = "tdx" ]; then
+    oc apply -f layeredimage-cm-tdx.yaml || exit 1
+elif [ "$TEE_TYPE" = "snp" ]; then
+    oc apply -f layeredimage-cm-snp.yaml || exit 1
+else
+    echo "Unsupported TEE_TYPE. It must be tdx or snp" || exit 1
+fi
```
Comment (gkurz) [inline scripts/install-helpers/baremetal-coco/install.sh]: `TEE_TYPE` is already validated by `verify_params()`. You can just use a `case` command as already done a few lines above.

```suggestion
    case $TEE_TYPE in
    tdx)
        oc apply -f layeredimage-cm-tdx.yaml || exit 1
        ;;
    snp)
        oc apply -f layeredimage-cm-snp.yaml || exit 1
        ;;
    esac
```

File: scripts/install-helpers/baremetal-coco/intel-dpo/sgx_device_plugin.yaml [hunk 26]
```diff
@@ -0,0 +1,14 @@
...
+kind: SgxDevicePlugin
+metadata:
+  name: sgxdeviceplugin-sample
+spec:
+  image: registry.connect.redhat.com/intel/intel-sgx-plugin@sha256:e54cdf61feb0b67e330ad24d92e7fa4000953c2ec4d4300d052e032d419d5e0a
+  enclaveLimit: 110
+  provisionLimit: 110
+  logLevel: 4
+  nodeSelector:
+    intel.feature.node.kubernetes.io/sgx: "true"
```
Comment (gkurz) [inline scripts/install-helpers/baremetal-coco/intel-dpo/sgx_device_plugin.yaml]: GH ui reports this file is lacking a terminating newline.

File: scripts/install-helpers/baremetal-coco/intel-dpo/install_operator.yaml [hunk 27]
```diff
@@ -0,0 +1,23 @@
...
+    operators.coreos.com/intel-device-plugins-operator.openshiftoperators: ""
+  name: intel-device-plugins-operator
+  namespace: openshift-operators
+spec:
+  channel: alpha
+  installPlanApproval: Automatic
+  name: intel-device-plugins-operator
+  source: certified-operators
+  sourceNamespace: openshift-marketplace
+  startingCSV: intel-device-plugins-operator.v0.29.1
```
Comment (gkurz) [inline scripts/install-helpers/baremetal-coco/intel-dpo/install_operator.yaml]: GH ui reports this file is lacking a terminating newline.

File: scripts/install-helpers/baremetal-coco/nfd/intel-rules.yaml [hunk 28]
```diff
@@ -0,0 +1,64 @@
+apiVersion: nfd.openshift.io/v1alpha1
+kind: NodeFeatureRule
+metadata:
+  name: intel-dp-devices
+  namespace: openshift-nfd
+spec:
+  rules:
+    - name: "intel.gpu"
+      labels:
+        "intel.feature.node.kubernetes.io/gpu": "true"
```
Comment (gkurz) [inline scripts/install-helpers/baremetal-coco/nfd/intel-rules.yaml]: This is supposed to be about CoCo, right ? Can you clarify  why we care for a GPU here ?

File: scripts/install-helpers/baremetal-coco/nfd/ns.yaml [hunk 29]
```diff
@@ -0,0 +1,7 @@
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: openshift-nfd
+  labels:
+    name: openshift-nfd
+    openshift.io/cluster-monitoring: "true"
```
Comment (gkurz) [inline scripts/install-helpers/baremetal-coco/nfd/ns.yaml]: GH ui reports this file is lacking a terminating newline.

File: scripts/install-helpers/baremetal-coco/nfd/og.yaml [hunk 30]
```diff
@@ -0,0 +1,9 @@
+apiVersion: operators.coreos.com/v1
+kind: OperatorGroup
+metadata:
+  generateName: openshift-nfd-
+  name: openshift-nfd
+  namespace: openshift-nfd
+spec:
+  targetNamespaces:
+  - openshift-nfd
```
Comment (gkurz) [inline scripts/install-helpers/baremetal-coco/nfd/og.yaml]: GH ui reports this file is lacking a terminating newline.

File: scripts/install-helpers/baremetal-coco/osc_catalog.yaml [hunk 31]
```diff
@@ -0,0 +1,12 @@
+apiVersion: operators.coreos.com/v1alpha1
+kind: CatalogSource
+metadata:
+  name: osc-upstream-catalog
+  namespace: openshift-marketplace
+spec:
+  displayName: OSC Upstream Operator Catalog
+  sourceType: grpc
+  image: quay.io/openshift_sandboxed_containers/openshift-sandboxed-containers-operator-catalog:1.8.1-1
```
Comment (gkurz) [inline scripts/install-helpers/baremetal-coco/osc_catalog.yaml]: Current (and probably GA) build is 1.8.1-3.
```suggestion
  image: quay.io/openshift_sandboxed_containers/openshift-sandboxed-containers-operator-catalog:1.8.1-3
```

**Review by gkurz — COMMENTED** [review]
Hi @bpradipt ! This looks pretty good to me. Posted some comments.

File: scripts/install-helpers/baremetal-coco/nfd/intel-rules.yaml (hunk 28 above)
Comment (bpradipt) [reply scripts/install-helpers/baremetal-coco/nfd/intel-rules.yaml]: Good point. I think this comes from confidential GPU. But right now it doesn't work with OSC anyway .. So I'll remove it

File: scripts/install-helpers/baremetal-coco/install.sh (hunk 10 above)
Comment (bpradipt) [reply scripts/install-helpers/baremetal-coco/install.sh:222]: This was added to keep shellcheck happy - https://www.shellcheck.net/wiki/SC2164

File: scripts/install-helpers/baremetal-coco/install.sh (hunk 22 above)
Comment (bpradipt) [reply scripts/install-helpers/baremetal-coco/install.sh]: Yeah, I'll make it consistent across the code.

File: scripts/install-helpers/baremetal-coco/install.sh (hunk 10 above)
Comment (bpradipt) [reply scripts/install-helpers/baremetal-coco/install.sh:222]: The return statements for both pushd and popfd were dictated by shellcheck. I can disable it as well, but thought it's better to keep it.  Let me know if you have a strong preference to remove it :-)

**Review by beraldoleal — COMMENTED** [review]
@bpradipt while reading this script, helm keeps popping on my mind. Have you considered using helm for this job? Any reason for not using helm here?

**Comment by bpradipt** [comment]
> @bpradipt while reading this script, helm keeps popping on my mind. Have you considered using helm for this job? Any reason for not using helm here?

No particular reason. We just happened to start with bash and that's all.

File: scripts/install-helpers/baremetal-coco/subs.yaml [hunk 32]
```diff
@@ -0,0 +1,12 @@
+apiVersion: operators.coreos.com/v1alpha1
```
Comment (beraldoleal) [inline scripts/install-helpers/baremetal-coco/subs.yaml]: since we have `subs-ga.yaml`, for better organization, maybe rename this one to `subs-upstream.yaml` ?

**Comment by beraldoleal** [comment]
> No particular reason. We just happened to start with bash and that's all.

Good to know.

I think we’d benefit a lot from using Helm. While this might not be the best place to start the effort, let’s keep the idea in mind and continue discussing it, as delaying could make migration harder later.

File: scripts/install-helpers/baremetal-coco/.gitignore (hunk 1 above)
Comment (bpradipt) [reply scripts/install-helpers/baremetal-coco/.gitignore]: Removed

**Comment by bpradipt** [comment]
> > No particular reason. We just happened to start with bash and that's all.
> 
> Good to know.
> 
> I think we’d benefit a lot from using Helm. While this might not be the best place to start the effort, let’s keep the idea in mind and continue discussing it, as delaying could make migration harder later.

We should also evaluate helm vs ansible approach imho.

File: scripts/install-helpers/baremetal-coco/install.sh (hunk 9 above)
Comment (bpradipt) [reply scripts/install-helpers/baremetal-coco/install.sh:218]: please see https://github.com/openshift/sandboxed-containers-operator/pull/479#discussion_r1869867440

File: scripts/install-helpers/baremetal-coco/install.sh (hunk 5 above)
Comment (gkurz) [reply scripts/install-helpers/baremetal-coco/install.sh]: This is still not printing `"$ip"`.

File: scripts/install-helpers/baremetal-coco/install.sh [hunk 33]
```diff
@@ -0,0 +1,728 @@
...
+    while [ $elapsed -lt "$timeout" ]; do
+        ip=$(oc get endpoints -n "$namespace" "$service" -o jsonpath='{.subsets[0].addresses[0].ip}')
+        if [ -n "$ip" ]; then
+            echo "Service $service IP is available"
+            return 0
+        fi
+        sleep $interval
+        elapsed=$((elapsed + interval))
+    done
+    echo "Service $service IP ($ip) is not available after $timeout seconds"
```
Comment (gkurz) [inline scripts/install-helpers/baremetal-coco/install.sh]: I think the `($ip)` should go to the success path.

File: scripts/install-helpers/baremetal-coco/install.sh (hunk 10 above)
Comment (gkurz) [reply scripts/install-helpers/baremetal-coco/install.sh:222]: I prefer to have shellcheck happy :wink:

File: scripts/install-helpers/baremetal-coco/install.sh (hunk 5 above)
Comment (bpradipt) [reply scripts/install-helpers/baremetal-coco/install.sh]: :-(

File: scripts/install-helpers/baremetal-coco/install.sh (hunk 33 above)
Comment (bpradipt) [reply scripts/install-helpers/baremetal-coco/install.sh]: yeah my bad.. Sorry about it

**Comment by bpradipt** [comment]
@gkurz @beraldoleal I think I have addressed all the comments. PTAL and let me know if I have missed something.
Also updated the SoB for the baremetal install commit based on original code in personal branch - https://github.com/bpradipt/coco-install

**Review by snir911 — APPROVED** [review]
LGTM thanks, maybe we can remove /hack completely now..

**Review by beraldoleal — APPROVED** [review]
LGTM

**Comment by bpradipt** [comment]
Force-pushed to fix incorrect usage of EOF when using `oc apply`. Thanks to @niteeshkd for fixing it.

**Comment by bpradipt** [comment]
@gkurz is it good to be merged ?

**Comment by gkurz** [comment]
> @gkurz is it good to be merged ?

Yes it is @bpradipt, I just want to understand why the e2e CI check is failing in this PR and not in #487 .
